virtualizarr/zarr.py

```diff
@@ -187,26 +213,13 @@ def _v3_codec_pipeline(self) -> Any:
             codec_configs.append(_num_codec_config_to_configurable(self.compressor))
 
         # convert the pipeline repr into actual v3 codec objects
-        codec_pipeline = parse_codecs(codec_configs)
-
-        return codec_pipeline
-
-    def serializer(self) -> Any:
-        """
-        testing
-        """
-        try:
-            from zarr.core.metadata.v3 import (  # type: ignore[import-untyped]
-                parse_codecs,
-            )
-        except ImportError:
-            raise ImportError("zarr v3 is required to generate v3 codec pipelines")
-        # https://github.com/zarr-developers/zarr-python/pull/1944#issuecomment-2151994097
-        # "If no ArrayBytesCodec is supplied, we can auto-add a BytesCodec"
-        bytes = dict(
-            name="bytes", configuration={}
-        )  # TODO need to handle endianess configuration
-        return parse_codecs([bytes])[0]
+        codecs = parse_codecs(codec_configs)
+        filters = v3_codecs_to_filters(codecs)
+        compressors = v3_codecs_to_compressors(codecs)
+        serializer = v3_codecs_to_serializer(codecs)
+        return ZarrV3Codecs(
+            filters=filters, compressors=compressors, serializer=serializer
+        )
```

Comment:
I'm curious if we replaced the `ZArray` class with zarr-python's `ArrayV3Metadata` would there be comparable functionality to parse out the filters, compressors and serializer, so those could be passed to zarr-python's `create_array` (same args as `require_array`)?

Reply:
I was going to do this with [`init_array`](https://github.com/zarr-developers/zarr-python/blob/main/src/zarr/core/array.py#L3796) but it requires a store path which i didnt want to deal with atm, but might be the right approach